[21231] apply table styling

HDinger opened the pull request:
started with the 'Admin/Users' table

https://community.openproject.org/work_packages/21231

HDinger committed:
Merge branch 'feature/18800-restyle-projects-table' into feature/21231-apply-table-styling

HDinger committed:
Merge branch 'feature/18800-restyle-projects-table' into feature/21231-apply-table-styling

HDinger committed:
started transforming the 'Admin/Users' table to new design

TeatroIO commented:
I've prepared a stage to preview changes. <a href='http://hdinger-feature-21231-apply-table-styling.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com' target='_blank'>Open stage</a> or <a href='http://hdinger-feature-21231-apply-table-styling.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com/teatro.log' target='_blank'>view logs</a>.

HDinger commented:
@ulferts can you please have a look at the 'sort_header_tag' and it's functionality? 
Thanks a lot!

oliverguenther commented:
@HDinger @ulferts I've encountered and addressed the issue with the LSG sort_header_tag in my own PR. This currently adds a second helper method to provide a column for LSG table headers, and the other should be removed once all tables are re-styled.

Maybe we should merge into a separate branch beside dev to have early access to the issues we both have to address (changed sort headers, re-working pagination style, ..)

https://github.com/oliverguenther/openproject/blob/feature/projects_pagination_and_sort/app/helpers/sort_helper.rb#L282-L291

ulferts committed:
adapt sort_header_tag helper

Follows LSG conventions now

ulferts committed:
apply adapted helper to users/index

ulferts commented:
I have now adapted the helper in https://github.com/HDinger/openproject/pull/1. As for the integration branch, I fear that the added complexity is too much. We simply need to get this feature into dev.

HDinger committed:
restyled the following tables: Admin/Groups, Admin/Roles, Admin/Types, Admin/Statuses, Admin/CustomFields

oliverguenther committed:
Fix cukes broken from changed table styles

oliverguenther commented:
https://github.com/HDinger/openproject/pull/2 should fix the broken cukes.

HDinger committed:
restyled the following tables: Admin/Info, Admin/Plugins, Admin/CustomFields, Admin/Enumerations, Admin/Colors, Admin/ProjectType, Admin/Auth, Workpackages/Summary

HDinger committed:
styled the folowing tables: Settings/Activities, Settings/Members, Settings/Categories, Settings/Versions, Settings/Forum, TimeSheets, StatusReporting, Forum

HDinger committed:
Merge pull request #2 from oliverguenther/fix/21231-fix-broken-cukes

Fix cukes broken from changed table styles

HDinger committed:
Merge pull request #1 from ulferts/feature/adapted_sort_header_tag_helper

Feature/adapted sort header tag helper

HDinger committed:
fixed 'copy-paste'-bug and used the 'sort-link'-tag again

HDinger committed:
fixed the broken specs

HDinger committed:
line breaks to please rubocop

HDinger committed:
Merge branch 'dev' into feature/21231-apply-table-styling

HDinger committed:
used the interactive-table-directive for background color of head when being hovered

ulferts committed:
introduce spacer to have a min width

The spacers have the same content as the visible header. As the visible
header has position abolute, it can not be used to determine the width,
but the spacer can.

ulferts committed:
handle invisible tables becoming visible

Because elements hidden via display:none do not take up space, it is
impossible to calculte their width prior to them becoming visible. This
poses a problem in the project settings where a lot of tables are only
reachable (become visible) after switching the tab.

HDinger committed:
styled the table Settings/Types

ulferts committed:
remove no longer required min-width rule

ulferts commented:
I addressed the problems you mentioned with regards to table headers on tabs not being their desired width in https://github.com/HDinger/openproject/pull/4 @HDinger

HDinger committed:
Merge pull request #4 from ulferts/fix/minimal_headers

Fix/minimal headers

oliverguenther commented:
With the new styling, the empty headers in the admin info table looks weird due the additional line from the System caption.
![bildschirmfoto 2015-08-14 um 14 52 41](https://cloud.githubusercontent.com/assets/459462/9274253/66267e1e-4294-11e5-98a7-2ab8866848cc.png)

In the plugins table, that line is also visible, but there is no line above it, it doesn't bother me so much.

Not sure If you agree, but if you do, I'd suggest either:
1. Add headers
2. Allow the header section to be hidden explicitly.

oliverguenther commented:
In the auth sources table, it's the other way around. Headers are visible, but no content.

What do you think of the following empty row notice?
![bildschirmfoto 2015-08-14 um 15 01 14](https://cloud.githubusercontent.com/assets/459462/9274353/5fc513f4-4295-11e5-9dec-1bc58372452e.png)

HDinger commented:
In my eyes the missing header doesn't look weird. @ulferts what do you think?
I like the solution for the case of missing content. For me this is a better way for missing content than the big notification that is displayed in all other tables.

oliverguenther commented:
On the restyled forums table, topics with no reply yet have a one-line row, while others with replies have two lines. This is low priority, but one could think about making the two-line by default, so it looks homogenous in most cases.

![bildschirmfoto_2015-08-14_um_16_06_37](https://cloud.githubusercontent.com/assets/459462/9275612/90eab430-429e-11e5-850a-3c2af12448bb.png)

HDinger committed:
styled more hidden tables in Admin/Workflows, Admin/Users, Admin/Groups, Admin/Roles

oliverguenther commented:
On org-config: 
1. Roles and permissions are overridden by `openproject-global_roles/app/views/roles/index.html.erb` and receive the old styling
2. The projects/settings/member table is modified by two hooks in `openproject-costs/app/views/hooks/costs/_view_projects_settings_members_table_{header,row}.html.erb` , which results in a wrong header placement.

![bildschirmfoto 2015-08-14 um 16 46 08](https://cloud.githubusercontent.com/assets/459462/9276401/00a90d30-42a4-11e5-8c27-335638c27f5a.png)

HDinger commented:
I can take care of the first problem.
The second problem is already fixed in an extra PR in the budget-plugin

https://github.com/finnlabs/openproject-costs/pull/153

HDinger committed:
fixed bug: one header for multipe columns

ulferts commented:
@HDinger the check/uncheck all functionality no longer works (project administration -> types):
![image](https://cloud.githubusercontent.com/assets/617519/9300445/4e26c7a4-44c6-11e5-97df-1ac0d04f22ec.png)

ulferts commented:
> In my eyes the missing header doesn't look weird. @ulferts what do you think?

I don't miss the headers. However, I think that it is wrong to use a table here (but it used to be this way already). The table is only taken for layouting IMO. We'd need to have a list similar to what we have on the project list:

![image](https://cloud.githubusercontent.com/assets/617519/9300540/69cca1b2-44c7-11e5-99d5-701fac1cdaaf.png)

However, I believe this is not necessarily within the scope of this PR.

ulferts commented:
> What do you think of the following empty row notice?

I believe we should stick to the large notice because it is used everywhere. Additionally, I would find it weird to have the information that the table holds no information as part of the table. Thinking of screen readers, I would be wondering, how they would read the information then. It would probably read "Name" > "Es wurden keine Authentifizierungs-Arten hinterlegt".

HDinger committed:
restyled the tables Boards, My, ProjectAssociation, Repositories, TimeEntries, Wiki/History, WorkPackages/ListSimple

oliverguenther commented:
You're right, the big notice is favorable. The notification was a quick idea using a colspan, but I admit that I do not know what screenreaders would output for a row with colspan over the whole table.

HDinger committed:
fixed bug: (un-)checkAll is working again in ProjectSettings/Types

HDinger committed:
the table Admin/Enumeration is now also styled by hand to make the site consistent

HDinger commented:
I fixed the 'Roles and Permissions'-Table in: https://github.com/finnlabs/openproject-global_roles/pull/41

HDinger committed:
fixed broken specs

oliverguenther commented:
> The table is only taken for layouting IMO. We'd need to have a list similar to what we have on the project list
> 
> However, I believe this is not necessarily within the scope of this PR.

I agree that this is out of scope. I'll refactor it into the attribute group when I extend the admin/information section for  [Feature #21275: Show remaining available storage information](https://community.openproject.org/work_packages/21275).

ulferts commented:
@HDinger I believe that there are still some styles on the repository table making it look slightly different (right align, text smaller):
![image](https://cloud.githubusercontent.com/assets/617519/9327415/f54562cc-45a0-11e5-8856-1612fb315559.png)
I think those rules can simply be removed.

HDinger committed:
removed unnecessary stylings

ulferts commented on app/views/admin/plugins.html.erb:
```diff
@@ -29,19 +29,42 @@ See doc/COPYRIGHT.rdoc for more details.
+    <div class="generic-table--no-results-container">
+      <h2 class="generic-table--no-results-title">
+        <i class="icon-info"></i>
+        Nothing to display
```
Those strings need to be internationalized (please also see the other tables). That means:
- adding a key to `config/en.yml` 
- using the key here: <%= l(:[Key]) %>

ulferts committed:
Merge branch 'dev' into pr/3343

ulferts commented on app/views/custom_fields/_index.html.erb:
```diff
@@ -26,36 +26,100 @@ Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+              <%= delete_link custom_field_path(custom_field) %>
+            </td>
+          </tr>
+        <% end; reset_cycle %>
```
the `reset_cycle` call is necessary only in combination with `cycle("odd", "even")` used for the colouring the table rows. As we don't do this any more, the call can be removed.

ulferts commented on app/views/enumerations/index.html.erb:
```diff
@@ -32,31 +32,75 @@ See doc/COPYRIGHT.rdoc for more details.
+        <div class="generic-table--header-background"></div>
+      </div>
+    </div>
     <% reset_cycle %>
```
please see above

ulferts commented on app/views/groups/_memberships.html.erb:
```diff
@@ -32,47 +32,82 @@ See doc/COPYRIGHT.rdoc for more details.
+                  </td>
+                </tr>
+              </tbody>
+            <% end; reset_cycle %>
```
please see above

ulferts commented on app/views/workflows/index.html.erb:
```diff
@@ -32,29 +32,47 @@ See doc/COPYRIGHT.rdoc for more details.
   <p class="nodata"><%= l(:label_no_data) %></p>
```
I believe this needs the proper styling as well

ulferts commented on app/views/workflows/index.html.erb:
```diff
@@ -32,29 +32,47 @@ See doc/COPYRIGHT.rdoc for more details.
+                <td><%= h type %></td>
+                <% roles.each do |role, count| -%>
+                  <td>
+                    <%= link_to((count > 0 ? count : image_tag('webalys/red_x.png')), {:action => 'edit', :role_id => role, :type_id => type}, :title => l(:button_edit)) %>
```
Not directly related to the PR (maybe do it in a separate one) but can we get rid of `image_tag('webalys/red_x.png')`. That way we can sever the connection to app/assets/images/webalys even more. 

I created a work package for it: https://community.openproject.org/work_packages/21312

ulferts commented on app/views/workflows/_form.html.erb:
```diff
@@ -26,48 +26,72 @@ Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+    </table>
+    <div class="generic-table--header-background"></div>
+  </div>
+</div>
```
Any chance we can get a different styling for this one?

It is currently looking like this:
![image](https://cloud.githubusercontent.com/assets/617519/9355757/1c151efc-467c-11e5-8268-24e472773afb.png)

I'd rather have the status headings (checkmark + name of status) more visible as headings. To that end I would like:
- to have the column highlighting work on the the status heading
- to have the status heading in bold and moved below the stronger line

Don't know if it is possible though.

ulferts commented:
Looks awesome @HDinger. I think we can merge this soon and I am really looking forward to it.

HDinger committed:
Merge pull request #5 from ulferts/feature/merge_dev_into_apply_table_styling

Feature/merge dev into apply table styling

HDinger committed:
fixed style bugs

HDinger committed:
fixed broken specs

HDinger committed:
Merge branch 'dev' into feature/21231-apply-table-styling

HDinger committed:
Merge branch 'dev' into feature/21231-apply-table-styling

ulferts commented:
@HDinger can you please try to again merge in dev now that #3397 is merged?

HDinger committed:
Merge branch 'dev' into feature/21231-apply-table-styling
